--- a/navi-codegen/src/scanner.rs
+++ b/navi-codegen/src/scanner.rs
@@ -293,7 +293,11 @@ fn assign_parents(routes: &mut Vec<RouteInfo>) {
             .unwrap_or(Path::new(""))
             .to_path_buf();
         if let Some(layout_id) = dir_to_layout.get(&search) {
-            route.parent = Some(layout_id.clone());
+            if layout_id != &route.route_id {
+                route.parent = Some(layout_id.clone());
+            } else {
+                route.parent = root_id.clone();
+            }
         } else {
             route.parent = root_id.clone();
         }
